### PR TITLE
Prepare 0.10.2 release

### PR DESCRIPTION
**Pull request by mtreinish**
This commit prepares for a 0.10.2 bugfix release to get the recent vf2
bugfixes on main out in the wild. Looking at the git log since 0.10.1
we've only merged fixes and requirements bumps since the release, so now
is a good time to push out the bugfix release before we start merging
new features.

### CODE REVIEW

**Comment by mtreinish** [comment]
Marking as on hold until https://github.com/Qiskit/retworkx/pull/435 and https://github.com/Qiskit/retworkx/pull/436 merge

File: releasenotes/notes/0.10.2/bugfix-vf2-parallel-edges-f8be125e255753f7.yaml [hunk 1]
```diff
@@ -3,12 +3,15 @@ fixes:
   - |
     Fixes the output of :func:`~retworkx.is_subgraph_isomorphic`
     if the input graphs have parallel edges.
+    Fixed `#429 <https://github.com/Qiskit/retworkx/issues/429`__
```
Comment (IvanIsCoding) [inline releasenotes/notes/0.10.2/bugfix-vf2-parallel-edges-f8be125e255753f7.yaml:6]: This line is making the docs build fail

**Review by IvanIsCoding — APPROVED** [review]
I agree, let's release 0.10.2 so we can start merging items for 0.11. We just need to fix one minor thing in the release notes and it should be good to go

File: releasenotes/notes/0.10.2/bugfix-vf2-parallel-edges-f8be125e255753f7.yaml (hunk 1 above)
Comment (mtreinish) [reply releasenotes/notes/0.10.2/bugfix-vf2-parallel-edges-f8be125e255753f7.yaml]: Oops my bad, fixed the typo in: https://github.com/Qiskit/retworkx/pull/434/commits/b7127af68b36dfcbcaa4d9396b4fb1b84742b083

File: releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+prelude: >
+  This release is a bugfix release that fixes some issues found in the 0.10.1
```
Comment (georgios-ts) [inline releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml:3]: btw, these bugs (except the empty input graph case) were present even in 0.9.0 release

File: releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  release. The fixes in this release are prrimarily in the vf2 implementation
```
Comment (georgios-ts) [inline releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml:4]:
```suggestion
  release. The fixes in this release are primarily in the vf2 implementation
```

File: releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml (hunk 2 above)
Comment (mtreinish) [reply releasenotes/notes/0.10.2/prepare-0.10.2-d75b2d86be66a6d9.yaml]: Good call, between this and #437 it's definitely more than just regressions in 0.10.x, I tweaked the wording here in https://github.com/Qiskit/retworkx/pull/434/commits/195fc43af1dc2932e2b0fa2f4fd89c7b5db12201

**Comment by mtreinish** [comment]
Let's wait on #439 before merging this, since it would be good to fix that while we're at it and there isn't anything critical to get 0.10.2 out right now.

**Pull request merged**